webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -462,6 +464,58 @@ func (v validator) validateNetworkInterfaceSpec(
 	return allErrs
 }
 
+// mtu and routes is available only with CloudInit bootstrap providers.
+// nameservers and searchDomains is available only with the following bootstrap
+// providers: CloudInit, LinuxPrep, and Sysprep (except for RawSysprep).
+func (v validator) validateNetworkSpecWithBootstrap(
+	interfacePath *field.Path,
+	interfaceSpec vmopv1.VirtualMachineNetworkInterfaceSpec,
+	vm *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+	cloudInit := vm.Spec.Bootstrap.CloudInit
+	linuxPrep := vm.Spec.Bootstrap.LinuxPrep
+	sysPrep := vm.Spec.Bootstrap.Sysprep
+
+	if mtu := interfaceSpec.MTU; mtu != nil && cloudInit == nil {
+		allErrs = append(allErrs, field.Invalid(
+			interfacePath.Child("mtu"),
+			"mtu",
+			"mtu is available only with the following bootstrap providers: CloudInit",
+		))
+	}
+
+	if routes := interfaceSpec.Routes; routes != nil && cloudInit == nil {
+		allErrs = append(allErrs, field.Invalid(
+			interfacePath.Child("routes"),
+			"routes",
+			"routes is available only with the following bootstrap providers: CloudInit",
+		))
+	}
+
+	if nameservers := interfaceSpec.Nameservers; nameservers != nil {
+		if cloudInit == nil && linuxPrep == nil && (!lib.IsWindowsSysprepFSSEnabled() || sysPrep == nil || !equality.Semantic.DeepEqual(sysPrep.RawSysprep, corev1.SecretKeySelector{})) {
```

Comment:
Since CloudInit, LinuxPrep, and Sysprep are mutually exclusive and the same as Sysprep and RawSysprep ( as enforced by the `validateBootstrap()` function above), can we simplify this statement to instead check if RawSysprep or VAppConfig is set?

Reply:
Bootstrap is entirely optional though. One suggestion to reduce line length & improve readability is to define a variable for the sysprep condition.